src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -287,23 +317,29 @@ public enum RisingRivals implements LogicCardInfo {
           resistance L, MINUS20
           pokeBody "Rainbow Float", {
             text "If any basic Energy card attached to Flygon is the same type as any of your Pokémon, the Retreat Cost for those Pokémon is 0."
-            delayedA {
+            getterA (GET_RETREAT_COST) { h->
+              if (!self.active && h.effect.target.owner == self.owner && h.effect.target.active && self.cards.filterByType(BASIC_ENERGY).find{h.effect.target.types.contains(it.basicType.first())}) {
+                h.object = 0
+              }
             }
           }
           move "Sand Wall", {
             text "40 damage. Discard a Stadium card your opponent has in play. If you do, prevent all effects of an attack, including damage, done to Flygon during your opponent’s next turn."
             energyCost C, C
-            attackRequirement {}
             onAttack {
-              damage 0
+              if(bg.stadiumInfoStruct && bg.stadiumInfoStruct.stadiumCard.player == self.owner.opposite){
+                preventAllEffectsNextTurn()
+                afterDamage {
+                  discard bg.stadiumInfoStruct.stadiumCard
```

Comment:
There is a static for doing this which may be a better idea if stadium stuff ever needs to be changed in the future.

Reply:
Idk if im just blind, but the only statics I see that interacts with stadiums are whirlwind 2, discardStadium and the stadium can be affected check. As this effect depends both on whether or not a stadium card is in play, and who played the stadium, I don't see how any of those would work here.

Reply:
You can keep the check, but discard using the discard static :P

I also know this effect is used a couple of times, you can probably make a static for the whole effect too.